politeiad/backendv2/tstorebe/tstorebe.go

```diff
@@ -1017,21 +1018,81 @@ func (t *tstoreBackend) Fsck() error {
 		return err
 	}
 
-	// Rebuild the inventory cache. This is a multi-step process:
-	//
-	// 1. Sort the tokens into two groups; unvetted and vetted. Each
-	//    group is additionally sorted by the timestamp of their most
-	//    recent status change.
-	//
-	// 2. Add the vetted tokens to the inventory cache. They MUST be
-	//    added starting with the oldest timestamp. The vetted token
-	//    is first added to the unvetted cache then moved to the vetted
-	//    cache. This is a limitation of the inventory API and mimicks
-	//    how records are added and updated when using the politeiad
-	//    API.
-	//
-	// 3. Add the unvetted tokens to the inventory cache. They MUST be
-	//    added starting with the oldest timestamp.
+	// Get the partial record for all tokens. This also guarantees that all
+	// tokens being manipulated actually correspond to a record on the backend.
+	records := make(map[string]*backend.Record, len(allTokens))
+	for _, token := range allTokens {
+		r, err := t.tstore.RecordPartial(token, 0, nil, true)
+		if err != nil {
+			return err
+		}
+		records[r.RecordMetadata.Token] = r
+	}
+
+	// Sort records into vetted and unvetted groups.
+	var (
+		vetted   []*backend.Record
+		unvetted []*backend.Record
+	)
```

Comment:
These variables had to be declared without a length max cap with make because of the `sort.Slice` function, which manipulates these slices. It panics with an invalid memory address access error when the slices have a fixed set length. Not sure exactly why by looking at the docs, still investigating on the reason, but this fixes the issue.

Reply:
Set the capacity, not the length.

`vetted = make([]*backend.Record, 0, len(allTokens))`

Reply:
oh right ofc. thx @lukebp